src/debugging.md

```diff
@@ -92,13 +93,14 @@ of the object.
 ## Debug mode assertions
 
 During development, you are highly encouraged to use Flutter's "debug"
-mode, sometimes referred to as "checked" mode. This is the default if
-you use `flutter run`. In this mode, the Dart `assert` statement is
-enabled, and the Flutter framework uses this to perform many runtime
-checks verifying that invariants are not being violated.
-
-When an invariant is violated, it is reported to the console, with
-some context information to help with tracking down the source of the
+mode. This is the default if you use `flutter run` or the bug icon
+in Android Studio. Some tools support assert statements through the
+command-line flag `--enable-asserts`.  In this mode,
+Dart `assert` statements are enabled, and the Flutter framework performs
```

Comment:
There is a danger that this could be misunderstood: 'performs runtime checks verifying that invariants are not being violated' sounds like we are checking language invariants (like heap soundness: "A variable of type T will never refer to an instance that cannot be typed as a T (but it can be an instance of a subtype)"), but language invariants are always maintained in Dart 2, not just when we run a program with `--enable-asserts`.

(OK, nobody talks about "trust-something" execution, and I'm pretty sure that we shouldn't mention it here, so we don't. ;-)

The only thing controlled by the `--enable-asserts` option is whether user-written `assert` statements are (1) ignored or (2) the argument is evaluated, and if `false` an `AssertionError` is thrown.

I don't know how to get the style right here, but maybe we could have something like 'In this mode, Dart `assert` statements are enabled, and the Flutter framework will then evaluate the argument to each `assert` statement encountered during execution, and throw if the result is `false`. This allows developers to enable or disable invariant checking, such that the associated performance cost is only paid during debugging sessions.'